test/nn/test_embedding.py

```diff
@@ -124,6 +124,35 @@ def test_embedding_bag_from_pretrained(self):
         output = embedding(input, torch.arange(input.size(0)))
         self.assertEqual(a, output)
 
+    @parametrize_test(
+        "bag_use_grad,per_sample_weights_use_grad",
+        [(True, True), (True, False), (False, True), (False, False)],
+    )
+    def test_embedding_bag_per_sample_weights_grad_cpu(
+        self, bag_use_grad: bool, per_sample_weights_use_grad: bool
+    ):
+        device = "cpu"
+        bag = torch.nn.EmbeddingBag(256, 256, mode="sum", device=device)
+        bag.requires_grad_(bag_use_grad)
+        x = torch.arange(1, 5, device=device).expand(3, -1)
+        w = torch.rand(3, 4, device=device, requires_grad=per_sample_weights_use_grad)
+        bag(x, per_sample_weights=F.softmax(w, dim=-1))
+
+    @parametrize_test(
+        "bag_use_grad,per_sample_weights_use_grad",
+        [(True, True), (True, False), (False, True), (False, False)],
+    )
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_embedding_bag_per_sample_weights_grad_cuda(
+        self, bag_use_grad: bool, per_sample_weights_use_grad: bool
+    ):
+        device = "cuda"
+        bag = torch.nn.EmbeddingBag(256, 256, mode="sum", device=device)
+        bag.requires_grad_(bag_use_grad)
+        x = torch.arange(1, 5, device=device).expand(3, -1)
+        w = torch.rand(3, 4, device=device, requires_grad=per_sample_weights_use_grad)
+        bag(x, per_sample_weights=F.softmax(w, dim=-1))
+
```

Comment:
Can you put this test under TestEmbeddingNNDeviceType? This lets us parametrize on device as well /avoid duplication between cpu/cuda.